lib/matplotlib/pyplot.py

```diff
@@ -1394,14 +1394,16 @@ def xticks(ticks=None, labels=None, **kwargs):
     """
     ax = gca()
 
-    if ticks is None and labels is None:
+    if ticks is None:
         locs = ax.get_xticks()
-        labels = ax.get_xticklabels()
-    elif labels is None:
+        if labels is not None:
+            raise AttributeError("Labels can't be set without setting ticks")
```

Comment:
IMHO [AttributeError](https://docs.python.org/3/library/exceptions.html#AttributeError) is not the right exception.

We only need the default value for `ticks` to support the getter-like behavior. Without that, we would have `def xticks(ticks, labels=None)`, which would raise `TypeError: xticks() missing 1 required positional parameter: 'ticks'`.

In that sense, I'd go with:
~~~
raise TypeError("xticks(): Parameter 'labels' can't be set without setting 'ticks'"
~~~

Reply:
agree with making this a typeerror (like for function signature mismatches)

Reply:
That makes sense. Fixing!